src/main/scala/chisel3/tester/ThreadedBackend.scala

```diff
@@ -32,7 +33,7 @@ trait ThreadedBackend {
 
   sealed trait HasOverridingPokes extends BaseTimescope {
     // List of timescopes overriding a signal, including ones closed / reverted on this timestep
-    val overridingPokes = mutable.HashMap[Data, mutable.ListBuffer[Timescope]]()
+    val overridingPokes: mutable.HashMap[Data, mutable.ListBuffer[Timescope]] = new mutable.HashMap
```

Comment:
All public methods should have explicit type annotations. Type inference may break encapsulation in these cases, because it depends on internal method and class details. Without an explicit type, a change to the internals of a method or val could alter the public API of the class without warning, potentially breaking client code. Explicit type annotations can also help to improve compile times.  - From scala-lang

Reply:
I see, though these aren't meant to be public APIs, they're strictly internal data structures, though not marked as such yet. Feel free to make these internal classes protected or something, if that's possible.
My preference is the rule of DRY is more important than strict adherence to some set of rules (which are of questionable applicability here), absent some overriding concern.

Reply:
I can make them private